dapps/marketplace/src/pages/create-listing/listing-types/FractionalListing/Location.js

```diff
@@ -0,0 +1,278 @@
+import React, { useState } from 'react'
+import { fbt } from 'fbt-runtime'
+import Geocode from 'react-geocode'
+
+import withIsMobile from 'hoc/withIsMobile'
+
+import Redirect from 'components/Redirect'
+import { DEFAULT_GOOGLE_MAPS_API_KEY } from 'constants/config'
+import Link from 'components/Link'
+import LocationMap from 'components/LocationMap'
+
+const ListingLocation = ({ prev, next, listing, onChange, isMobile }) => {
+  const isEditingListing = /\/edit\//.test(location.href)
+  const [redirect, setRedirect] = useState(false)
+  const [fetchingLocation, setFetchingLocation] = useState(false)
+  const [forceEditLocation, setForceEditLocation] = useState(false)
+  const [formLocationValue, setFormLocationValue] = useState(
+    listing.locationAddress || ''
+  )
+  const [formLocationError, setFormLocationError] = useState(null)
+  const [currentLocationError, setCurrentLocationError] = useState(null)
+
+  const locationInput = () => {
+    let className = 'px-3'
+    if (formLocationError) {
+      className += ' is-invalid'
+    }
+
+    return {
+      value: formLocationValue,
+      className,
+      name: 'location',
+      onChange: e => {
+        setFormLocationValue(e.target.value)
+        setFormLocationError(null)
+      }
+    }
+  }
```

Comment:
Does this have to be a function? I see this being called once once. Calling a function each time a component renders seems like a unwanted (still negligible) overhead.

Reply:
It needs to be a function so when the `formLocationError` state changes, it adds the error class to the html node.

Reply:
I might be wrong here. But changing that function definition (and it's call in the next line) to something like should work right? 
Or am I missing something here?
```
    let className = 'px-3'
    if (formLocationError) {
      className += ' is-invalid'
    }

    const input = {
      value: formLocationValue,
      className,
      name: 'location',
      onChange: e => {
        setFormLocationValue(e.target.value)
        setFormLocationError(null)
      }
    }
```

The function gets called all the time and it only populates an object that inside `input`, AFAICT. Why not populate the object directly instead of a function call?

Reply:
Ah yes, I don't know what I was thinking. You are right thanks!